### DISCUSSION TITLE
webview em vez de url launcher quando possível?

### DISCUSSION THREAD

**Discussion by ivofernandes** [olmps/memo#176] [Geral (Português)]
Que acham de abrir webview em vez de url launcher quando possível?

Se quiserem posso fazer esse commit e PR porque já fiz em projectos meus, é fazer um if(!kIsWeb) abrir web view else abrir url

Bem simples e assim não enviam assim logo o user para fora da app

**Comment by matuella** [comment]
Antes de mais nada, muito obrigado pela sugestão @ivofernandes!

Eu acredito que enviar o usuário "para fora do app" não é algo necessariamente ruim, no entanto acredito ser um ponto completamente discutível.

Eu vejo que a abordagem de hoje nos isenta de termos que abraçar essa responsabilidade de controlar Webviews (e adicionar novas dependências) com um custo muito baixo, pois a experiência do usuário é muito similar.

Existem alguns pontos também, tal como, se _embeddarmos_ uma Webview dentro da aplicação, não usaremos o browser de preferência do usuário. Vejo que ambas abordagens (Webview ou usando o browser) são usadas nos mais diversos apps, mas não vejo uma linha muito clara dos pontos positivos de uma ou outra, quando estamos falando de links externos. É claro que, em casos de necessitar um controle maior do conteúdo disposto, uma Webview é uma excelente (se não única) solução, no entanto, não acredito ser nosso caso hoje.

Queria saber mais da sua opinião, você acredita que abrir um link externo tem um peso relevante a ponto de necessitar de uma Webview dentro do app?

**Reply by ivofernandes** [reply]
De facto é algo muito longe de uma ciência exata.

Acho que a experiência só muda caso o link sugerido seja de facto útil, se o utilizador acabar a navegar numa webview tem sempre a app a um clique de distância, caso abra um browser e navegue no browser como volta à app? Faz back 4 vezes? Vai ao menu de apps abertas? Talvez se distraia no browser e já nem volte? Mas a webview acaba a poder/dever ter um atalho para abrir num browser, só fica mais difícil de sair do app e se perder nos confins da internet
